### DISCUSSION TITLE
How do I use it?

### DISCUSSION THREAD

**Discussion by VictimOfPing** [acheong08/ChatGPT#1184] [Q&A]
I would like to use it but I don't understand how it works

**Comment by hrizal** [comment]
in my opinion, basically this is a way to access chatgpt using your account (easier and more flexible).

with this script, you can create your own API to connect with various interfaces (mobile apps, web apps or chat apps)

I've tried it by wrapping it in php to call this python script, and it runs smoothly


todo:

1. `python -m pip install --upgrade revChatGPT`
2. create python script as on below, and save as cgpt-api.py

```
import sys
import json
from revChatGPT.V1 import Chatbot

c = {"access_token":"-","conversation_id":"-", "parent_id":"-"}
c["conversation_id"] = sys.argv[2]
c["parent_id"] = sys.argv[3]
c["model"] = "gpt-4"
c["access_token"] = sys.argv[4]

chatbot = Chatbot(config=c)

prev_text = ""
response = ""

if sys.argv[2]=="-":
   sys.argv[2]=""

if sys.argv[3]=="-":
   sys.argv[3]=""

for data in chatbot.ask(
    sys.argv[1]+chr(13)+chr(10), sys.argv[2], sys.argv[3] 
):
    response = data["message"]

resp = {"msg":"-", "chat_id":"-", "parent_id":"-","detail":"-" }
if "message" in data:
    resp["msg"] = data["message"]
resp["chat_id"] = data["conversation_id"]
resp["parent_id"] = data["parent_id"]
if "detail" in data:
    resp["detail"] = data["detail"]

print(json.dumps(resp))

```

and call from command line or php shell like 

`python3 cgpt-api.py "your question"  conversation_id  parent_id  your_access_token`

if you still have question, you can ask to chatgpt too hehehe
